src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,73 @@
+package duke.task;
+
+/**
+ * Represents a task with a description, status, and type.
+ * This is an abstract class that serves as a base for specific task types like ToDo, Deadline, and Event.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+    protected TaskType taskType;
```

Comment:
Is it really necessary to have a label of subtype? the printing of type information can be handled in the subclass.